src/LobbyServer.hs

```diff
@@ -119,26 +119,29 @@ getGamesList remoteGames = do
   return $ map fst gameList
 
 -- |Lets a player join a 'LobbyGame'
-playerJoinGame :: Server ConcurrentClientList -> Server GamesList -> String -> Server ()
+playerJoinGame :: Server ConcurrentClientList -> Server GamesList -> String -> Server Bool
 playerJoinGame remoteClientList remoteGameList gameID = do
   clientList <- remoteClientList >>= liftIO . CC.readMVar
   gameList <- remoteGameList
   sid <- getSessionID
-  case lookupClientEntry sid clientList of
-    Just player -> liftIO $ CC.modifyMVar_ gameList $
-      \gList -> return $ addPlayerToGame player gameID gList
-
-    _ -> return ()
-
-  maybeGame <- findGameWithSid gameList
+  maybeGame <- liftIO $ findGameWithID gameID gameList
   case maybeGame of
-    Just (_,gameData) -> liftIO $ messageClients PlayerJoinedGame (players gameData)
-    nothing           -> return ()
+    Nothing           -> return False
+    Just (_,gameData) ->
+      if maxAmountOfPlayers gameData > length (players gameData) then do
+          case lookupClientEntry sid clientList of
+            Nothing     -> error "playerJoinGame: Client not registered"
```

Comment:
Throwing an error will crash the server. I did the same while working with chats but solved it by creaing an CharError the the ChatMessage datatype which i then send to the client. Maybe we should generalize this even more and create a LobbyError in the LobbyMessage datatype which we can use as a kind of error report to the client. What do you think?